File: bibliotheque-backend/src/main/java/com/ibizabroker/bibliotheque/entity/Reservation.java
package com.ibizabroker.bibliotheque.entity;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import lombok.Data;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import jakarta.persistence.*;
import java.util.Date;

@Data
@Entity
@EntityListeners(AuditingEntityListener.class)
@Table(name = "reservation")
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Integer bookId;
    private Integer userId;

    /**
     * Nom d'un livre PAS encore enregistré — transporté par la requête de
     * réservation puis résolu/créé côté service. Jamais persisté (sinon
     * une colonne sans rapport apparaîtrait sur la table reservation).
     */
    @Transient
    private String newBookName;

    @Enumerated(EnumType.STRING)
    private StatutReservation statut;

    @Temporal(TemporalType.TIMESTAMP)
    @JsonSerialize(using = JsonDataSerializer.class)
    private Date dateReservation;

    @Temporal(TemporalType.TIMESTAMP)
    @JsonSerialize(using = JsonDataSerializer.class)
    private Date dateExpiration;
}

File: bibliotheque-frontend/src/app/_model/reservation.ts
export enum StatutReservation {
  /** Demande soumise par un adhérent — le personnel doit l'accepter. */
  DEMANDE = 'DEMANDE',
  EN_ATTENTE = 'EN_ATTENTE',
  DISPONIBLE = 'DISPONIBLE',
  ANNULEE = 'ANNULEE',
  EXPIREE = 'EXPIREE',
  HONOREE = 'HONOREE'
}

export class Reservation {
  id: number;
  bookId: number;
  /** Nom d'un livre PAS encore enregistré — le backend le crée à 0 exemplaire puis le réserve. */
  newBookName?: string;
  userId: number;
  statut: StatutReservation;
  dateReservation: Date;
  dateExpiration: Date;
}
